### PR TITLE
feat : support for volta

### LINKED ISSUE #4262: Volta support
**What would you like Renovate to be able to do?**
Much like renovate upgrades node in a `.nvmrc` file, it would also be nice to upgrade it in the volta property in the package.json file too.

**Describe the solution you'd like**
Whenever yarn, or node gets an upgrade within the ranges outlined in a config or something, to also upgrade it in the package.json file in the volta config.

**Describe alternatives you've considered**
Well right now, I just use a bash script to sync the volta properties from `.nvmrc`, but yarn I have to do manually. 

**Additional context**
See [volta.sh](https://volta.sh/) for more details or https://github.com/volta-cli/volta

### PR DESCRIPTION
**Pull request by nihalwasheretoo**
Added support for volta, get's managed by npm.
Docs: https://docs.volta.sh/guide/understanding

Closes #4262

### CODE REVIEW

**Comment by nihal9ns** [comment]
@rarkins  good example to test against:

`.nvmrc `

```
8.9.2
```

`package.json`

```
{
  "name": "volta-test",
  "version": "1.0.0",
  "description": "Test project for volta",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "engines": {
    "node": "8.9.2"
  },
  "author": "Nihal",
  "license": "ISC",
  "volta": {
    "node": "8.9.2",
    "yarn": "1.12.3"
  }
}
```

**Comment by nihal9ns** [comment]
Currently volta only supports pinning `node` and `yarn` in a project

File: lib/config/definitions.ts [hunk 1]
```diff
@@ -510,6 +510,15 @@ const options: RenovateOptions[] = [
     mergeable: true,
     cli: false,
   },
+  {
```
Comment (rarkins) [inline lib/config/definitions.ts]: If volta is managed by `npm` then there's no need for this. Remove it completely as well as the doc entry for it. Instead, add a description within the npm manager. Alternatively, implement it as a new manager instead (lib/manager/volta).

File: lib/manager/npm/extract/index.ts [hunk 2]
```diff
@@ -181,6 +182,25 @@ export async function extractPackageFile(
...
+        dep.datasource = 'github';
+        dep.lookupName = 'nodejs/node';
+        dep.versionScheme = 'node';
+      } else if (depName === 'yarn') {
+        dep.datasource = 'npm';
+        dep.commitMessageTopic = 'Yarn';
+      } else {
+        dep.skipReason = 'unknown-volta';
+      }
+      if (!isValid(dep.currentValue)) {
```
Comment (rarkins) [inline lib/manager/npm/extract/index.ts]: What happens if they specify a version like `1.10` for yarn?

**Review by rarkins — CHANGES_REQUESTED** [review]
Either make it a separate manager or fully merge it under npm

File: lib/config/definitions.ts (hunk 1 above)
Comment (nihal9ns) [reply lib/config/definitions.ts]: I'll merge it under npm.

File: lib/manager/npm/extract/index.ts (hunk 2 above)
Comment (nihal9ns) [reply lib/manager/npm/extract/index.ts]: It updates it to 1.17

File: lib/config/definitions.ts (hunk 1 above)
Comment (nihal9ns) [reply lib/config/definitions.ts]: I just didn't understand this part ->  `Instead, add a description within the npm manager` 
You mean add a description under npm config in definitions? or docs?

Comment (rarkins) [reply lib/config/definitions.ts]: I mean in the docs, under `## npm` or whatever it is

File: lib/manager/npm/extract/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/manager/npm/extract/index.ts]: You've tested this working? I wasn't sure if we considered 1.10 to be a valid value

File: lib/config/definitions.ts (hunk 1 above)
Comment (nihal9ns) [reply lib/config/definitions.ts]: okay, got it!

File: lib/manager/npm/extract/index.ts (hunk 2 above)
Comment (nihal9ns) [reply lib/manager/npm/extract/index.ts]: Yes, tested it against my test repo for volta.

Screen Shot 2019-09-23 at 8 51 59 AM

File: docs/usage/configuration-options.md [hunk 3]
```diff
@@ -560,6 +560,19 @@ Check out our [Node.js documentation](https://docs.renovatebot.com/node) for a c
 
 ## npm
 
+Extend this if you wish to configure rules specifically for `volta` definitions. Currently only `node` and `yarn` are supported. Example:
```
Comment (rarkins) [inline docs/usage/configuration-options.md]: Not correct. You extend the `npm` if you want to configure rules for `npm`. `volta` is now just a `depType`. I think let's remove this text completely for now and add an issue to properly document `npm` here in future (including volta, engines, etc).

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/configuration-options.md (hunk 3 above)
Comment (nihal9ns) [reply docs/usage/configuration-options.md]: I don't disagree

**Pull request merged**